src/presentational/AssignmentForm.jsx

```diff
@@ -0,0 +1,144 @@
+import { Component, PropTypes } from 'react';
+import { Link } from 'react-router';
+
+import InputElement from './InputElement';
+
+const initialState = {
+  classifications_target: '',
+  description: '',
+  duedate: '',
+  name: '',
+  students: [],
+  subjects: [],
+}
+
+
+class AssignmentForm extends Component {
+  constructor(props) {
+    super(props);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.renderStudentList = this.renderStudentList.bind(this);
+    this.toggleStudent = this.toggleStudent.bind(this);
+    this.state = Object.assign({}, initialState);
+  }
+
+
+  toggleStudent(id) {
+    if (this.state.students.includes(id)) {
+      this.state.students = this.state.students.filter(student => student !== id)
+      this.setState({
+        students: this.state.students,
+      })
+    } else {
+      this.state.students.push(id);
+      this.setState({
+        students: this.state.students,
+      })
+    }
+  }
+
+  handleChange(e) {
+    let nextState = {};
```

Comment:
I imagine previously you were doing something like

```
const x = { foo };
x = { bar };
```

`const` means it should always point to the same reference, but the object it refers to is allowed to change. So here, you're changing a property on `nextState`, which is legit. _Replacing_ the entire state with a new state object is not